### PR TITLE
Map opengl toolbar buttons

### PR DESCRIPTION
**Pull request by orange-vertex**
Make the toolbar buttons work for the OpenGL view and some minor fixes here and there

### CODE REVIEW

File: depthmapX/depthmapView.cpp [hunk 1]
```diff
@@ -217,7 +217,7 @@ bool QDepthmapView::eventFilter(QObject *object, QEvent *e)
 					if (val == -1.0f)
 						setToolTip("No value");
 					else if (val != -2.0f)
-						setToolTip(QString("%1").arg(val));
```
Comment (blsemo) [inline depthmapX/depthmapView.cpp]: What's up with the spacing here? Are you replacing whitespace with tabs?

File: depthmapX/glview.cpp [hunk 2]
```diff
@@ -36,29 +36,45 @@ GLView::GLView(QWidget *parent, QGraphDoc* doc, const QRgb &backgroundColour, co
...
-    if(m_pDoc->m_meta_graph->getViewClass() & m_pDoc->m_meta_graph->VIEWVGA) {
+    if(m_pDoc->m_meta_graph->getViewClass() & MetaGraph::VIEWVGA) {
         m_visiblePointMap.loadGLObjects(m_pDoc->m_meta_graph->getDisplayedPointMap());
     }
 
-    if(m_pDoc->m_meta_graph->getViewClass() & m_pDoc->m_meta_graph->VIEWDATA) {
+    if(m_pDoc->m_meta_graph->getViewClass() & MetaGraph::VIEWDATA) {
         m_visibleDataMap.loadGLObjects(m_pDoc->m_meta_graph->getDisplayedDataMap());
     }
+    PafColor pafForegroundColour;
```
Comment (blsemo) [inline depthmapX/glview.cpp]: Is it technically necessary to go via PafColour here? Or just to be in line with dX idioms? Maybe we should break PafColor out of vertex.h/cpp and make it a first class member of the either sala or genlib?

File: depthmapX/mainwindow.cpp [hunk 3]
```diff
@@ -1815,29 +1818,39 @@ void MainWindow::SelectButtonTriggered()
 {
     m_selected_mapbar_item = ID_MAPBAR_ITEM_SELECT;
     activeQDepthmapView()->OnEditSelect();
+    GLView* glView = getFirstGLView();
```
Comment (blsemo) [inline depthmapX/mainwindow.cpp]: Not sure I like this (or maybe I don't understand?): Can't this code know whether we're in QDepthMapView or GLViewMode and only do the one that's actually used? This whole "go through widgets and see if one is the view I want to work on" pattern seems somewhat insane, and this is going in the same direction, if I am not mistaken?

Comment (orange-vertex) [reply depthmapX/mainwindow.cpp]: The following (and final in this saga) pull request actually has a main class "MapView" that both QDepthmapView and GLView descend from (And the above piece becomes like [this](https://github.com/orange-vertex/depthmapX/blob/map_opengl_default_window/depthmapX/mainwindow.cpp#L1838) so for MainWindow it doesn't matter which one you're looking at, as long as it's a MapView

Yes, the pattern is insane, and it is jumping through weird hoops, but I wasn't aiming to fix it, as that would require going into the table, scatter and agent views as well...

**Pull request merged**